src/main/java/frc/robot/RobotContainer.java

```diff
@@ -123,42 +139,115 @@ public void burnFlashSparks() {
    * joysticks}.
    */
   private void configureBindings() {
-    driverController.a().onTrue(new InstantCommand(drive::toggleSkids));
-    driverController.b().onTrue(new OuttakeSequence(lift));
-    driverController.x().onTrue(new InstantCommand(lift::cancelScore, lift));
+    driverController
+        .b()
+        .whileTrue(
+            new OuttakeSequence(lift, lights)
+                .finallyDo(
+                    (boolean interrupted) -> {
+                      lift.home();
+                    }));
+    driverController.a().onTrue(new InstantCommand(lift::cancelScore, lift));
+    driverController
+        .x()
+        .onTrue(
+            new InstantCommand(
+                () -> {
+                  drive.throttle(Cal.SwerveSubsystem.THROTTLE_FOR_SLOW_BUTTON);
+                }));
+    driverController
+        .x()
+        .onFalse(
+            new InstantCommand(
+                () -> {
+                  drive.throttle(1.0);
+                }));
+    driverController.y().whileTrue(new RunCommand(drive::setX, drive));
 
     driverController
-        .y()
-        .onTrue(new InstantCommand(() -> lift.ManualPrepScoreSequence(lights), lift));
+        .rightBumper()
+        .onTrue(
+            new InstantCommand(
+                () -> {
+                  lift.ManualPrepScoreSequence(lights);
+                },
+                lift));
 
     driverController.back().onTrue(new InstantCommand(lift::home, lift));
-    driverController.start().onTrue(new InstantCommand(drive::halfSpeedToggle));
 
-    // TODO Maybe: steal
-    // TODO: implement autoscore command for teleop
-    // driverController.rightBumper().onTrue(new InstantCommand(lift::prepScore, lift));
+    driverController.start().onTrue(new InstantCommand(drive::resetYaw));
+
+    driverController
+        .leftBumper()
+        .onTrue(
+            new SequentialCommandGroup(
+                new InstantCommand(lift::openGrabber),
+                new InstantCommand(
+                    () -> {
+                      lift.setDesiredPosition(LiftPosition.SHELF);
+                    }),
+                new WaitUntilCommand(
+                    () -> {
+                      return lift.seeGamePiece();
+                    }),
+                new InstantCommand(lift::closeGrabber),
+                new WaitCommand(0.2),
+                new InstantCommand(
+                    () -> {
+                      lift.setDesiredPosition(LiftPosition.STARTING);
+                    },
+                    lift)));
```

Comment:
This should probably be its own distinct command, rather than being written inline here.

Reply:
Sure moved